Source/mpqapi.cpp

```diff
@@ -574,28 +566,20 @@ void CloseMPQ(const char *pszArchive, BOOL bFree, int dwChar)
 
 void mpqapi_store_modified_time(const char *pszArchive, int dwChar)
 {
-	int v2;                                // esi
-	const char *v3;                        // ebx
-	HANDLE v4;                             // eax
-	int v5;                                // esi
-	struct _WIN32_FIND_DATAA FindFileData; // [esp+8h] [ebp-1E0h]
-	char dst[160];                         // [esp+148h] [ebp-A0h]
-
-	v2 = dwChar;
-	v3 = pszArchive;
+	HANDLE handle;
+	struct _WIN32_FIND_DATAA FindFileData;
+	char dst[160];
+
 	if (gbMaxPlayers != 1) {
 		mpqapi_reg_load_modification_time(dst, 160);
-		v4 = FindFirstFile(v3, &FindFileData);
-		if (v4 != INVALID_HANDLE_VALUE) {
-			FindClose(v4);
-			v5 = 16 * v2;
-			*(_DWORD *)&dst[v5 + 8] = FindFileData.ftLastWriteTime.dwLowDateTime;
-			*(_DWORD *)&dst[v5 + 12] = FindFileData.ftLastWriteTime.dwHighDateTime;
+		handle = FindFirstFile(pszArchive, &FindFileData);
+		if (handle != INVALID_HANDLE_VALUE) {
+			FindClose(handle);
+			*((FILETIME*) (dst) + dwChar * 2 + 1) = FindFileData.ftLastWriteTime;

```

Comment:
As such, dst is to be understood as an array of time stamps, one per character (hero of the game), and each time stamps consists of two DWORDs, `ftCreationTime` and `ftLastWriteTime`; thus the use of `2*dwChar` and `2*dwChar + 1`. This data is later XORed and stored under a registry key "Video Player" to avoid recognition. Likely, the data is used to help prevent tampering of save files.

```suggestion
			*((DWORD *)(dst) + dwChar * 2 + 1) = FindFileData.ftLastWriteTime;
```

Reply:
@Predelnik, before merging, just double-check that it's still bin-exact with this suggestion applied. I think it should be, but can't verify at the moment.